internal/rsl/rsl.go

```diff
@@ -0,0 +1,273 @@
+	if _, err := os.Stat(refPath); err != nil {
+		if os.IsNotExist(err) {
+			if err := os.WriteFile(refPath, plumbing.ZeroHash[:], 0644); err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+
+	return nil
+}
+
+type EntryType interface {
+	Commit(bool) error
+	createCommitMessage() string
+}
+
+type Entry struct {
+	RefName  string
+	CommitID plumbing.Hash
+}
+
+// NewEntry returns an Entry object for a normal RSL entry.
+func NewEntry(refName string, commitID plumbing.Hash) *Entry {
+	return &Entry{RefName: refName, CommitID: commitID}
+}
+
+// Commit creates a commit object in the RSL for the Entry.
+func (e Entry) Commit(sign bool) error {
+	repo, err := common.GetRepositoryHandler()
+	if err != nil {
+		return err
+	}
+
+	return gitinterface.Commit(repo, plumbing.ZeroHash, RSLRef, e.createCommitMessage(), sign)
+}
+
+func (e Entry) createCommitMessage() string {
+	lines := []string{
+		EntryHeader,
+		"",
+		fmt.Sprintf("ref: %s", e.RefName),
+		fmt.Sprintf("commit: %s", e.CommitID.String()),
+	}
+	return strings.Join(lines, "\n")
+}
+
+type Annotation struct {
+	RSLEntryIDs []plumbing.Hash
+	Skip        bool
+	Message     string
+}
+
+// NewAnnotation returns an Annotation object that applies to one or more prior
+// RSL entries.
+func NewAnnotation(rslEntryIDs []plumbing.Hash, skip bool, message string) *Annotation {
+	return &Annotation{RSLEntryIDs: rslEntryIDs, Skip: skip, Message: message}
+}
+
+// Commit creates a commit object in the RSL for the Annotation.
+func (a Annotation) Commit(sign bool) error {
+	repo, err := common.GetRepositoryHandler()
+	if err != nil {
+		return err
+	}
+
+	// Check if referred entries exist in the RSL namespace.
+	for _, id := range a.RSLEntryIDs {
+		if _, err := GetEntry(id); err != nil {
+			return err
+		}
+	}
+
+	return gitinterface.Commit(repo, plumbing.ZeroHash, RSLRef, a.createCommitMessage(), sign)
+}
+
+func (a Annotation) createCommitMessage() string {
+	lines := []string{
+		AnnotationHeader,
+		"",
+	}
+
+	for _, entry := range a.RSLEntryIDs {
+		lines = append(lines, fmt.Sprintf("entryID: %s", entry.String()))
+	}
+
+	if a.Skip {
+		lines = append(lines, "skip: true")
+	} else {
+		lines = append(lines, "skip: false")
+	}
+
+	if len(a.Message) != 0 {
+		lines = append(lines, BeginMessage)
+		lines = append(lines, strings.Split(a.Message, "\n")...)
+		lines = append(lines, EndMessage)
+	}
+
+	return strings.Join(lines, "\n")
+}
+
+// GetLatestEntry returns the latest entry available locally in the RSL.
+// TODO: There is no information yet about the signature for the entry.
+func GetLatestEntry() (EntryType, error) {
+	repo, err := common.GetRepositoryHandler()
+	if err != nil {
+		return nil, err
+	}
+
+	ref, err := repo.Reference(plumbing.ReferenceName(RSLRef), true)
+	if err != nil {
+		return nil, err
+	}
+
+	commitObj, err := repo.CommitObject(ref.Hash())
+	if err != nil {
+		return nil, err
+	}
+
+	return parseRSLEntryText(commitObj.Message)
```

Comment:
You may want to consider refactoring this so this just calls GetEntry. This way any changes in entry parsing can be consolidated there. This is just a special case where we want to get the first value.

Reply:
Agreed but we'd be doing 80% of the work to get the entry ID to pass to GetEntry.

Reply:
And GetEntry would then redo getting the reference...